src/modules/service-provider/service-provider-api.module.ts

```diff
@@ -17,7 +18,8 @@ import { PersonModule } from '../person/person.module.js';
         RolleModule,
         PersonModule,
     ],
-    providers: [StreamableFileFactory],
+    providers: [StreamableFileFactory, ServiceProviderService],
+    exports: [ServiceProviderService],
```

Comment:
Why the export here? If another module needs this service, it should be exported (and provided) in the `ServiceProviderModule`

Reply:
I did not know how to deal with a Circular Dependency and this was my workaround. I referenced [this section in the docs](https://docs.nestjs.com/fundamentals/circular-dependency) and added `forwardRef` to the `RolleModule` and the `ServiceProviderModule` to fix it. Is that the right way to do it or do we have a different way?

Fixed in [5596065](https://github.com/dBildungsplattform/dbildungs-iam-server/pull/648/commits/55960656bfa01f7dbb6cd2368742605f7e14b81c)

Reply:
If it works that's great, we had a lot of trouble with circular dependencies in the past that couldn't be solves with `forwardRef`